lib/rubocop/cop/lint/space_inside_string_interpolation.rb

```diff
@@ -0,0 +1,40 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for whitespace within string interpolations.
+      #
+      # Good:
+      #    var = "This is the #{good} example"
+      #
+      # Bad:
+      #    var = "This is the #{ bad } example"
+      #
+      class SpaceInsideStringInterpolation < Cop
+        MSG = 'Space inside string interpolation detected.'
+        def on_dstr(node)
+          node.children.select { |n| n.type == :begin }.each do |begin_node|
+            final_node = begin_node.children.last
+
+            interp = final_node.loc.expression
+            if range_with_surrounding_space(interp) != interp
+              add_offense(final_node, :expression, MSG)
```

Comment:
Like many restaurants, RuboCop uses MSG by default. So you can call `add_offense` with just the first two arguments.